content/articles/creating-a-shopping-cart-api/index.md

```diff
@@ -0,0 +1,588 @@
+    List<Product> checkAllTheProductsInTheCartByCustomer(Long id);
+    List<Product> checkAllTheProductsInTheCartByAdmin(Long id);
+    BigDecimal calculatePrice();
+    int totalNumberOfItems();
+    BigDecimal calculatingPurchase(Long id);
+}
+}
+```
+#### DataService
+```java
+public interface DataService {
+    Customer registerCustomer(Customer customer);
+    Admin registerAdmin(Admin admin);
+    List<Customer> gettingAllCustomers();
+    void removeCustomer(Long id);
+    void removeAdmin(Long id);
+    List<Admin> getAllAdmin();
+}
+}
+```
+Now You will see how we code to the interface. Any method that is in the interface must be implemented in the implementation. In this code, I used the `@Service` annotation in the implementation. We have the following implementation of the two classes in our interface and they are:
+#### CartServiceImpl
+```java
+@Service
+public class CartServiceImpl implements cartService{
+    @Autowired
+    private CustomerDao customerRepo;
+    @Autowired
+    private productDao productRepo;
+    @Autowired
+    private Cart cart;
+    @Autowired
+    private AdminDao adminDao;
+
+    @Override
+    public void addProductToCartByCustomer(Long id, Product product) {
+if(customerRepo.getCustomer(id)!=null){
+
+    cart.addToTheCart(product);
+}
+    }
+
+    @Override
+    public void addProductToCartByAdmin(Long id, Product product) {
+        if(adminDao.getAdmin(id)!=null){
+            cart.addToTheCart(product);
+        }
+    }
+
+    @Override
+    public void removeProductsFromCartByCustomer(Long id, Product product) {
+if(customerRepo.getCustomer(id)!=null){
+    cart.removeFromCart(product);
+}
+    }
+
+    @Override
+    public void removeProductsFromCartByAdmin(Long id, Product product) {
+        if(adminDao.getAdmin(id)!=null){
+            cart.removeFromCart(product);
+        }
+    }
+
+    @Override
+    public List<Product> checkAllTheProductsInTheCartByCustomer(Long id) {
+        List<Product> productList=new ArrayList<>();
+        if (customerRepo.getCustomer(id) != null) {
+            productList=cart.displayAllItems();
+        }
+            return productList;
+        }
+
+    @Override
+    public List<Product> checkAllTheProductsInTheCartByAdmin(Long id) {
+        List<Product> productList=new ArrayList<>();
+        if (adminDao.getAdmin(id) != null) {
+            productList=cart.displayAllItems();
+        }
+        return productList;
+    }
+
+
+    @Override
+    public BigDecimal calculatePrice() {
+        return cart.calculatePriceInCart();
+    }
+
+
+    @Override
+    public int totalNumberOfItems() {
+        return cart.totalNumberOfProducts();
+    }
+
+    @Override
+    public BigDecimal calculatingPurchase(Long id) {
+        return customerRepo.getCustomer(id).getBalance().subtract(calculatePrice());
+    }
+}
+```
+N/B In this code snippet I used descriptive terms to enable you to understand the function of the methods.
```

Comment:
Not sure this is worthy of note. This just sounds awkward here. I would remove this.